src/bijectors/coupling_layer.jl

```diff
@@ -0,0 +1,244 @@
+using SparseArrays
+
+"""
+    PartitionMask{A}(A_1::A, A_2::A, A_3::A) where {A}
+
+This is used to partition and recombine a vector into 3 disjoint "subvectors".
```

Comment:
Why do we partition inputs into 3 disjoint subspaces? It is to support "multi-scale" flow or something?

Reply:
You split into 3 parts (x₁, x₂, x₃) s.t. you can then define a mapping b(x₁; θ(x₂)) and x₃ isn't used for anything. Alternatively, you could just define θ(x₂, x₃) (i.e. only split into 2 parts) but then you again have to do a "filtering" on indices inside θ. And we can make it so that if you do `PartitionMask(A_1, A_2)`, then the last part is just an empty array (we already have such a constructor based on indices directly, i.e. `PartitionMask(n::Int, indices1, indices2)`)

Reply:
I see. It makes sense. Thanks.